From f4d5f68d90ad006f1e2b9f9ca014ffc6b1e8aa13 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 5 Oct 2021 10:02:26 +0200
Subject: [PATCH 428/529] Input: goodix - add delay after reset

This delay is necessary to avoid an i2c error after reset
sequence.

Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: Ic5957dea4330cc33cc7aeb5749d1d42b7378698a
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/228187
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/292530
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 drivers/input/touchscreen/goodix.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1407,6 +1407,9 @@ reset:
 				dev_err(&client->dev, "Gpio unreset failed.\n");
 				return error;
 			}
+
+			/* need a delay after reset to test I2C */
+			msleep(100);
 		}
 	}
 
